src/plugins/vis_builder/public/embeddable/vis_builder_embeddable.tsx

```diff
@@ -79,6 +81,7 @@ export class VisBuilderEmbeddable extends Embeddable<VisBuilderInput, VisBuilder
   private subscriptions: Subscription[] = [];
   private node?: HTMLElement;
   private savedVis?: VisBuilderSavedVis;
+  private savedObject?: VisBuilderSavedObject;
```

Comment:
instead of passing the saved object down, cant you modify the `getStateFromSavedObject` function to return the `searchSourceFields` as a part of `savedVis` too.

```ts
export interface VisBuilderSavedVis
  extends Pick<
    VisBuilderSavedObjectAttributes,
    'id' | 'title' | 'description' | 'searchSourceFields'
  > {
  state: RenderState;
}
```

Reply:
Yeah we could. let me polish the code.